### PR TITLE
feat(gatby-script): Shim requestIdleCallback

### PR DESCRIPTION
**Pull request by marvinjude**
## Description
Shim requestIdleCallback, cancelIdleCallback 

[sc-48843]

### CODE REVIEW

File: e2e-tests/development-runtime/cypress/integration/gatsby-script/shimmed-req-idle-callback.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+// The page that we will assert against
+const page = `/scripts-with-sources`
+
+Cypress.on(`window:before:load`, win => {
+  cy.spy(win, "requestIdleCallback").as("requestIdleCallback")
+  win.requestIdleCallback = undefined
+})
+
+beforeEach(() => {
```
Comment (tyhopp) [inline e2e-tests/development-runtime/cypress/integration/gatsby-script/shimmed-req-idle-callback.js:12]: Remove this, not needed

File: e2e-tests/development-runtime/cypress/integration/gatsby-script/shimmed-req-idle-callback.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { Script, scripts } from "../../scripts"
+import { ResourceRecord } from "../../records"
+
+// The page that we will assert against
+const page = `/scripts-with-sources`
```
Comment (tyhopp) [inline e2e-tests/development-runtime/cypress/integration/gatsby-script/shimmed-req-idle-callback.js:5]: Should be `/gatsby-script-scripts-with-sources` now

File: e2e-tests/development-runtime/cypress/integration/gatsby-script/shimmed-req-idle-callback.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { Script, scripts } from "../../scripts"
```
Comment (tyhopp) [inline e2e-tests/development-runtime/cypress/integration/gatsby-script/shimmed-req-idle-callback.js:1]: Import paths are wrong

File: packages/gatsby-script/src/request-idle-callback-shim.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+// https://developer.chrome.com/blog/using-requestidlecallback/#checking-for-requestidlecallback
+// https://github.com/vercel/next.js/blob/canary/packages/next/client/request-idle-callback.ts
+
+export const requestIdleCallback =
```
Comment (tyhopp) [inline packages/gatsby-script/src/request-idle-callback-shim.ts:4]: Why is this not imported in `gatsby-script`?

**Review by tyhopp — CHANGES_REQUESTED** [review]
Few changes, please make sure the tests pass locally and in CI.

You can update the outdated load plugin snapshots with `yarn jest /Users/<YOUR_USER>/Developer/gatsby/gatsby/packages/gatsby/src/bootstrap/load-plugins/__tests__ --updateSnapshot` so that unit tests pass and you can see these e2e tests pass in CI.

File: packages/gatsby-script/src/request-idle-callback-shim.ts (hunk 4 above)
Comment (marvinjude) [reply packages/gatsby-script/src/request-idle-callback-shim.ts:4]: Ouch! probably missed a commit while cherry-picking.

**Pull request merged**